### ISSUE TITLE
XOAUTH error despite disabled on Gmail

### ISSUE THREAD

**Issue by cataclysmic** [OfflineIMAP/offlineimap#360]
#### General informations

- offlineimap version (`offlineimap -V`):  7.0.2  / also 6.6.1
- Python version:  3.5
- server name or domain:  Gmail
- CLI options:  none

#### Configuration file offlineimaprc

[general]

accounts= Private,Work

[Account Private]
localrepository = LocalPriv
remoterepository = RemotePriv

[Account Work]
localrepository = LocalWork
remoterepository = RemoteWork

[Repository LocalPriv]
type = Maildir
localfolders = ~/.mail/priv

[Repository LocalWork]
type = Maildir
localfolders = ~/.mail/work

[Repository RemotePriv]
type = IMAP
ssl = yes
sslcacertfile = /etc/ssl/certs/ca-certificates.crt
reference = Mail
remotehost = imap.gmail.com
remoteuser = user_mail@gmail.com
autorefresh = 5

[Repository RemoteWork]
type = IMAP
ssl = yes
sslcacertfile = /etc/ssl/certs/ca-certificates.crt
reference = Mail
remotehost = imap.gmail.com
remoteuser = user_mail@gmail.com
autorefresh = 5

######Log output ###
OfflineIMAP 7.0.2
  Licensed under the GNU GPL v2 or any later version (with an OpenSSL exception)
Account sync Private:
 *** Processing account Private
 Establishing connection to imap.gmail.com:993
 XOAUTH2 authentication failed: [AUTHENTICATIONFAILED] Invalid credentials (Failure)
Enter password for account 'RemotePriv': 
 Creating new Local Status db for LocalPriv:
 *** Finished account 'Private' in 0:05

Idential for calling 'work' account.

#### Steps to reproduce the error

- execute "offlineimap" 
- twoway authorization disabled on Gmail

**Comment by nicolas33** [comment]
Tune the `auth_mechanisms` to workaround this. However, I'm surprised since XOAUTH2 won't be tried if not configured, here.

**Comment by cataclysmic** [comment]
I am especially baffled
        as "Access for less secure applications" in Gmail is turned on
        ...
    
    
    On 28.07.2016 17:54, Nicolas Sebrecht
      wrote:
    
    
      Tune the auth_mechanisms to workaround this.
        However, I'm surprised since XOAUTH2 won't be tried if not
        configured, here.
      —
        You are receiving this because you authored the thread.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.

**Comment by nicolas33** [comment]
It's not Gmail fault. It's offlineimap trying XOAUTH2 while no configuration option is enabled in the configuration file.

**Comment by nicolas33** [comment]
Are you sure you did not set `oauth2_refresh_token` and `oauth2_access_token`?

**Comment by cataclysmic** [comment]
That's my offlineimaprc file as is.
    
    
    On 28.07.2016 18:07, Nicolas Sebrecht
      wrote:
    
    
      Are you sure you did not set oauth2_refresh_token
        and oauth2_access_token?
      —
        You are receiving this because you authored the thread.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.

**Comment by nicolas33** [comment]
Does it work with Py2?

**Comment by cataclysmic** [comment]
Tried running offlineimap
      6.6.1 (ubuntu install) again:
      
      cataclysmic@musashi:~$ offlineimap 
      OfflineIMAP 6.6.1
        Licensed under the GNU GPL v2 or any later version (with an
      OpenSSL exception)
      Account sync Private:
       *** Processing account Private
       Establishing connection to imap.gmail.com:993
       XOAUTH2 authentication failed: AUTHENTICATE command error: BAD
      ['Client aborted AUTHENTICATE command. x188mb381559940wmb']. Data:
      DMJC2 AUTHENTICATE XOAUTH2
      
      Enter password for account 'RemotePriv': 
       Creating folder Mail/[RemotePriv]
       *** Finished account 'Private' in 0:12
      Account sync Work:
       *** Processing account Work
       Establishing connection to imap.gmail.com:993
       XOAUTH2 authentication failed: AUTHENTICATE command error: BAD
      ['Client aborted AUTHENTICATE command. w76mb252224741wmd']. Data:
      ENCN2 AUTHENTICATE XOAUTH2
      
      Enter password for account 'RemoteWork': 
       *** Finished account 'Work' in 0:05
      
      (Deleted the offlineimap folder before.)
      
      Is there a separate config file somewhere other then
      .offlineimaprc ?
      
      Especially the weird part is, yesterday it worked. I was setting
      up notmuch and had 2-3 trial runs that worked fine.
      The laptop had been in sleep mode since then, could that mess up
      something?
      
    
    On 28.07.2016 18:39, Nicolas Sebrecht
      wrote:
    
    
      Does it work with Py2?
      —
        You are receiving this because you authored the thread.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.

**Comment by nicolas33** [comment]
> Tried running offlineimap
>       6.6.1 (ubuntu install) again:

Sorry, I meant v7.0.2 or `next` branch with py2.



> Especially the weird part is, yesterday it worked. I was setting
>       up notmuch and had 2-3 trial runs that worked fine.

Weird. There is only offlineimaprc and authentication does not makes use of the cache.

>       The laptop had been in sleep mode since then, could that mess up
>       something?

It should not. offlineimap should not even try XOAUTH authentication when both the refresh and access token are not configured. With v6.6.1 and prior, XOAUTH2 should not be used if the access token is not configured.

Since I'm pluzzed, I'm curious if a reboot could help.

**Comment by cataclysmic** [comment]
I got it to work by
        setting "auth_mechanisms = PLAIN,
        LOGIN". After that I didn't get an error anymore.
    But it didn't download
        mails. I then removed "reference = Mail" and it started
        downloading.
    Still weird that it
        tried to use XOAUTH despite being not configured. I am currently
        on the default ubuntu package 6.6.1.
    I will test 7.0.2
        stable branch later today.
    Thank you for your
        help. ^_^
    
    
    On 29.07.2016 02:23, Nicolas Sebrecht
      wrote:
    
    
      
        Tried running offlineimap
          6.6.1 (ubuntu install) again:
      
      Sorry, I meant v7.0.2 or next branch with py2.
      
        Especially the weird part is, yesterday it worked. I was
          setting
          up notmuch and had 2-3 trial runs that worked fine.
      
      Weird. There is only offlineimaprc and authentication does not
        makes use of the cache.
      
          The laptop had been in sleep mode since then, could that mess up
  something?

      
      It should not. offlineimap should not even try XOAUTH
        authentication when both the refresh and access token are not
        configured. With v6.6.1 and prior, XOAUTH2 should not be used if
        the access token is not configured.
      Since I'm pluzzed, I'm curious if a reboot could help.
      —
        You are receiving this because you authored the thread.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.

**Comment by nicolas33** [comment]
I'd try the current `next` branch or wait for the upcoming v7.0.3. Thanks for the feedback.

**Comment by nicolas33** [comment]
See #362, too.

**Comment by nicolas33** [comment]
I'm closing because it becomes hard to follow. If you still have issues, open a new bug report and strip uneeded parts next time, please.

**Comment by tommasop** [comment]
I have the exact same issue with version 7.0.14 and setting auth_mechanisms works as a workaround

**Comment by nicolas33** [comment]
@tommasop Could you open a new issue, please?

**Comment by tommasop** [comment]
Done [https://github.com/OfflineIMAP/offlineimap/issues/442](https://github.com/OfflineIMAP/offlineimap/issues/442)
